app/shared/session/useLoginErrors.ts

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from "react";
+import { useSearchParams } from "react-router-dom";
+import { sentryException } from "../sentry";
+
+export const AUTH_PROVIDERS = {
+  LOGIN_DEV: "login_dev",
+  LOGIN_GITHUB: "login_github",
+  LOGIN_GOOGLE: "login_google",
+};
+
+export const LOGIN_ERROR_MESSAGES = {
+  [AUTH_PROVIDERS.LOGIN_DEV]: "There has been an issue logging you in with dev",
+  [AUTH_PROVIDERS.LOGIN_GITHUB]:
+    "There has been an issue logging you in with Github",
+  [AUTH_PROVIDERS.LOGIN_GOOGLE]:
+    "There has been an issue logging you in with Google",
+};
+
+export const useLoginErrors = (): string => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [messageToReturn, setMessageToReturn] = useState("");
+  const error = searchParams.get("error");
+  const message = searchParams.get("message");
+
+  useEffect(() => {
+    const hasMessageToShow =
+      error !== null && Boolean(message) && typeof message === "string";
```

Comment:
Why Boolean check, are you checking if message is empty?

Reply:
otherwise typeof check should be enough

Reply:
so if you really checking for empty, why not `message !== ''` that's literally saying what you want to do, otherwise its quite obfuscated

Reply:
updated